src/ListGroup.js

```diff
@@ -9,6 +9,11 @@ class ListGroup extends React.Component {
       (item, index) => cloneElement(item, { key: item.key ? item.key : index })
     );
 
+    if (this.areCustomChildren(items)) {
+      let componentClass = this.props.componentClass;
+      return componentClass === 'ul' ? this.renderUL(items) : this.renderDiv(items);
```

Comment:
Tbh I didn't know you could specify the componentClass itself in the oneOf propType, etc. Fixing.